From d281511bb67d5c922b511e0afd9fd3c8adc93274 Mon Sep 17 00:00:00 2001
From: Ben Wolsieffer <benwolsieffer@gmail.com>
Date: Sat, 3 Oct 2020 19:21:30 -0400
Subject: [PATCH] HACK: remove conflict detection because it is broken with
 IPv6

---
 avahi-core/server.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/avahi-core/server.c b/avahi-core/server.c
index e507750..48a3994 100644
--- a/avahi-core/server.c
+++ b/avahi-core/server.c
@@ -224,7 +224,7 @@ static void incoming_probe(AvahiServer *s, AvahiRecord *record, AvahiInterface *
         }
     }
 
-    if (!ours) {
+    if (!ours && 0) {
         char *t = avahi_record_to_string(record);
 
         if (won)
@@ -318,7 +318,7 @@ static int handle_conflict(AvahiServer *s, AvahiInterface *i, AvahiRecord *recor
         }
     }
 
-    if (!ours && conflict) {
+    if (!ours && conflict && 0) {
         char *t;
 
         valid = 0;
-- 
2.28.0

